component/wallet-cli/pkg/walletrunner/wallet_runner_oidc4ci.go

```diff
@@ -229,11 +232,192 @@ func (s *Service) RunOIDC4CI(config *OIDC4CIConfig, hooks *Hooks) error {
 	return nil
 }
 
+var matchRegex = regexp.MustCompile(oidc4ci.WalletInitFlowClaimRegex)
+
+func extractIssuerUrlFromScopes(scopes []string) (string, error) {
+	for _, scope := range scopes {
+		if matchRegex.MatchString(scope) {
+			return scope, nil
+		}
+	}
+
+	return "", errors.New("issuer url not found in scopes")
+}
+
+func (s *Service) RunOIDC4CIWalletInitiated(config *OIDC4CIConfig, hooks *Hooks) error {
+	log.Println("Starting OIDC4VCI authorized code flow Wallet initiated")
+	ctx := context.Background()
+	// Check whether scope contains combined string Issuer URL||ClaimEndpoint||credentialTemplateID
+	issuerUrl, err := extractIssuerUrlFromScopes(config.Scope)
+	if err != nil {
+		return errors.New(
+			"undefined scopes supplied. " +
+				"Make sure one of the provided scope is an vcs issuer url format. ref " +
+				oidc4ci.WalletInitFlowClaimRegex)
+	}
+
+	oidcIssuerCredentialConfig, err := s.getIssuerCredentialsOIDCConfig(
+		issuerUrl,
+	)
+	if err != nil {
+		return fmt.Errorf("get issuer oidc issuer config: %w", err)
+	}
+
+	oidcConfig, err := s.getIssuerOIDCConfig(ctx, issuerUrl)
+	if err != nil {
+		return fmt.Errorf("get issuer oidc config: %w", err)
+	}
+
+	redirectURL, err := url.Parse(config.RedirectURI)
+	if err != nil {
+		return fmt.Errorf("parse redirect url: %w", err)
+	}
+
+	var listener net.Listener
+
+	if config.Login == "" { // bind listener for callback server to support log in with a browser
+		listener, err = net.Listen("tcp4", "127.0.0.1:0")
+		if err != nil {
+			return fmt.Errorf("listen: %w", err)
+		}
+
+		redirectURL.Host = fmt.Sprintf(
+			"%s:%d",
+			redirectURL.Hostname(),
+			listener.Addr().(*net.TCPAddr).Port,
+		)
+	}
+
+	s.oauthClient = &oauth2.Config{
+		ClientID:    config.ClientID,
+		RedirectURL: redirectURL.String(),
+		Scopes:      config.Scope,
+		Endpoint: oauth2.Endpoint{
+			AuthURL:   oidcConfig.AuthorizationEndpoint,
+			TokenURL:  oidcConfig.TokenEndpoint,
+			AuthStyle: oauth2.AuthStyleInHeader,
+		},
+	}
+
+	issuerState := uuid.New().String()
```

Comment:
We shouldn't need an issuer state for the wallet-initiated flow, right?

Reply:
i am not aware of the requirement for removing fields, the only change which was requested for wallet flow is additional scope. cc @sudeshrshetty

Reply:
The issuer state should come from the issuer, and without a credential offer, the wallet won't know what the issuer state is. Also, why would a random UUID created by the client be accepted by the issuer?

Reply:
As i can see currently, this field is **required** in openapi schema for this endpoint. https://github.com/trustbloc/vcs/blob/main/docs/v1/openapi.yaml#L626  @sudeshrshetty

Reply:
"Also, why would a random UUID created by the client be accepted by the issuer" - because for wallet-initiated flow we have different logic on backend, the transaction is created on /callback step, that's why any valid string works here

Reply:
issuer state is OPTIONAL field in spec. Not required in case of wallet initiated flow.